Upgrade Drupal to 9.2.6 and related contributed modules

richardbporter opened the pull request:
Resolves https://github.com/uiowa/uiowa/issues/4163
Resolves https://github.com/uiowa/uiowa/issues/4195
Resolves #4173 

Test the functionality addressed by the patches removed: https://github.com/uiowa/uiowa/compare/drupal-nine-two?expand=1#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34

Only core has update hooks.

<details>
<summary>Notable core diffs</summary>

- Replacements for deprecated, updates due to dependencies
- The tour module previously depended on jQuery Joyride. Joyride is now replaced with Shepherd. The core/jquery.joyride is now deprecated and core/shepherd is added. Tour config has changed, it no longer uses attributes with a data-id or data-class to indicate a target element for the tour. The selector property is used instead.
A BC layer is provided so that existing tours will still function, and any themes which use Stable or Stable9 as a base will have consistent markup for the Tour so that no theming changes should be necessary.
- CKEditor has been updated
- Sortable has been updated
- QuickEdit is being moved to contrib (death sentence?)
- Support for YouTube playlists

</details>

Contrib updates:
- search_api_autocomplete
- field_group
- layout_builder_lock
- metatag
- entity_embed
- core-composer-scaffold
- media_entity_twitter
- weight
- tvi
- views_ical
- smart_date

# ToDo

- [x] Fix sppa.uiowa.edu site split
  - _"There are some weird config issues going on with this split, but @pyrello said to ignore until we have a better idea of any changes in core and until we update config_split."_ - https://github.com/uiowa/uiowa/pull/4015
- [x] Fix pharmacy.uiowa.edu/views.view.people_list_block
- [x] Fix 500 error on https://grad.local.drupal.uiowa.edu/academics/thesis-defenses
  - I think it is choking on the number of events. out of the box the module does not seem to support https://fullcalendar.io/docs/validRange but when i limit by view filter, it starts to render again.
- [x] Run in DEV

# To Test

- Look for new errors
- Look for config diff issues
- Check for duplicate drag handles
- Media types (e.g. remote video) can not use all of the media view modes.
- You can still nest/change parent for menu links using node edit form
- Check admissions full nodes for article, person, student profile (entity print, moderation config updates)
- Check admissions that you can still properly weight scholarship content
- Check admissions that colleges taxonomy terms are still displaying like they are in PROD (tvi)
- Check grad full nodes for person, scholarship (metatag, person type status)
- Check grad thesis defense functionality as it relates to smart_date
- Check policy.clas for any issues with views_ical
- Check genetics people view menu links?
- Check immuno.grad research areas
- You can add a youtube playlist and have it display... OK.
- check authentication and caching in DEV

Commit:
Update entity_embed to 1.2.0

Commit:
Remove patch and update drupal core to  9.1.13

Commit:
Upgrading drupal/mysql56 (1.1.0 => 1.2.0)

Commit:
Remove fix duplicate drag handles in claro patch

Commit:
Update core to 9.2

 - Upgrading composer/semver (3.2.2 => 3.2.5)
  - Upgrading doctrine/annotations (1.11.1 => 1.13.1)
  - Upgrading drupal/core (9.1.13 => 9.2.6)
  - Upgrading drupal/core-recommended (9.1.13 => 9.2.6)
  - Upgrading egulias/email-validator (2.1.22 => 2.1.25)
  - Upgrading guzzlehttp/promises (1.4.0 => 1.4.1)
  - Upgrading guzzlehttp/psr7 (1.7.0 => 1.8.2)
  - Upgrading laminas/laminas-diactoros (2.5.0 => 2.6.0)
  - Upgrading laminas/laminas-feed (2.13.0 => 2.14.1)
  - Upgrading laminas/laminas-stdlib (3.3.0 => 3.3.1)
  - Upgrading laminas/laminas-zendframework-bridge (1.1.1 => 1.2.0)
  - Upgrading pear/pear_exception (v1.0.1 => v1.0.2)
  - Upgrading psr/container (1.0.0 => 1.1.1)
  - Upgrading psr/log (1.1.3 => 1.1.4)
  - Upgrading symfony/console (v4.4.19 => v4.4.25)
  - Upgrading symfony/debug (v4.4.19 => v4.4.25)
  - Upgrading symfony/dependency-injection (v4.4.19 => v4.4.25)
  - Locking symfony/deprecation-contracts (v2.4.0)
  - Upgrading symfony/error-handler (v4.4.19 => v4.4.25)
  - Upgrading symfony/event-dispatcher (v4.4.19 => v4.4.25)
  - Upgrading symfony/http-client-contracts (v2.3.1 => v2.4.0)
  - Upgrading symfony/http-foundation (v4.4.19 => v4.4.25)
  - Upgrading symfony/http-kernel (v4.4.19 => v4.4.25)
  - Upgrading symfony/mime (v5.1.11 => v5.3.0)
  - Upgrading symfony/polyfill-ctype (v1.20.0 => v1.23.0)
  - Upgrading symfony/polyfill-iconv (v1.20.0 => v1.23.0)
  - Upgrading symfony/polyfill-intl-idn (v1.20.0 => v1.23.0)
  - Upgrading symfony/polyfill-intl-normalizer (v1.20.0 => v1.23.0)
  - Upgrading symfony/polyfill-mbstring (v1.20.0 => v1.23.0)
  - Upgrading symfony/polyfill-php80 (v1.20.0 => v1.23.0)
  - Upgrading symfony/process (v4.4.19 => v4.4.25)
  - Upgrading symfony/psr-http-message-bridge (v2.0.2 => v2.1.0)
  - Upgrading symfony/routing (v4.4.19 => v4.4.25)
  - Upgrading symfony/serializer (v4.4.19 => v4.4.25)
  - Upgrading symfony/service-contracts (v2.2.0 => v2.4.0)
  - Upgrading symfony/translation (v4.4.19 => v4.4.25)
  - Upgrading symfony/translation-contracts (v2.3.0 => v2.4.0)
  - Upgrading symfony/validator (v4.4.19 => v4.4.25)
  - Upgrading symfony/var-dumper (v5.1.11 => v5.3.0)
  - Upgrading symfony/yaml (v4.4.19 => v4.4.25)
  - Upgrading twig/twig (v2.14.1 => v2.14.6)

Commit:
Commit 9.2 scaffold file changes

richardbporter committed:
Merge branch 'main' into drupal-nine-two

richardbporter committed:
Merge branch 'main' into drupal-nine-two

Commit:
Revert "Disable drupal sec checks"

This reverts commit 2072c47cc128dbf38598849b903a26cc43bcb0b6.

richardbporter committed:
Merge branch 'main' into drupal-nine-two

joewhitsitt committed:
update tour config

joewhitsitt committed:
update local env tour

joewhitsitt committed:
update google tag and webform config

joewhitsitt committed:
update cshs to 8.x-3.5

joewhitsitt committed:
Merge branch 'main' into drupal-nine-two

joewhitsitt committed:
update search_api_autocomplete, field_group, layout_builder_lock, metatag

joewhitsitt committed:
update admissions config

joewhitsitt committed:
Merge branch 'main' into drupal-nine-two

joewhitsitt committed:
update grad config

joewhitsitt committed:
Merge branch 'drupal-nine-two' of github.com:uiowa/uiowa into drupal-nine-two

joewhitsitt committed:
Merge branch 'main' into drupal-nine-two

joewhitsitt committed:
merge main

joewhitsitt committed:
update core-composer-scaffold to 9.2.6

joewhitsitt committed:
update media_entity_twitter to 2.7.0

joewhitsitt committed:
update weight to 3.3.0

joewhitsitt committed:
Merge branch 'main' into drupal-nine-two

joewhitsitt committed:
update tvi 1.0.0-rc4

joewhitsitt committed:
Merge branch 'drupal-nine-two' of github.com:uiowa/uiowa into drupal-nine-two

joewhitsitt committed:
Merge branch 'main' into drupal-nine-two

joewhitsitt committed:
update views_ical 1.0.0-alpha8

joewhitsitt committed:
Merge branch 'main' into drupal-nine-two

joewhitsitt committed:
update smart_date 3.4.2

joewhitsitt committed:
Merge branch 'main' into drupal-nine-two

pyrello committed:
Merge branch 'main' into drupal-nine-two

joewhitsitt committed:
resolves #4195

joewhitsitt committed:
Merge branch 'drupal-nine-two' of github.com:uiowa/uiowa into drupal-nine-two

joewhitsitt committed:
update ga config

pyrello committed:
Merge branch 'main' into drupal-nine-two

pyrello commented on composer.json:
```diff
@@ -227,12 +227,10 @@
-                "Media viewmode control": "https://www.drupal.org/files/issues/2020-01-31/3097416-ckeditor-view-modes-8.9-combined.patch",
                 "Expose Layout Builder data to REST and JSON:API": "https://www.drupal.org/files/issues/2020-07-07/2942975-116.patch",
-                "Themes have no context of the entity being rendered in preprocessing a layout when using Layout builder": "https://www.drupal.org/files/issues/2020-02-11/3111192-entity-14.patch",
+                "Themes have no context of the entity being rendered in preprocessing a layout when using Layout builder": "https://git.drupalcode.org/project/drupal/-/merge_requests/608.patch",
```
If we want to use this patch, we should create an actual patch file and upload it to the issue, since people could continue to update this MR branch and there is nothing here locking it to the current commit.

pyrello replied:
I will work on creating a patch for this.

pyrello replied:
This is fixed.

pyrello commented on config/default/linkit.linkit_profile.default.yml:
```diff
@@ -16,7 +16,7 @@ matchers:
     weight: -10
     settings:
-      metadata: 'created by [node:author] | updated [node:changed]'
+      metadata: '[node:content-type] - /node/[node:nid] created by [node:author] | updated [node:changed:html_date]'
```
Was this a change that came from a module config update?

joewhitsitt replied:
rogue feature enhancement - https://github.com/uiowa/uiowa/issues/4195

pyrello commented on config/sites/admissions.uiowa.edu/core.entity_view_display.node.article.default.yml:
```diff
@@ -0,0 +1,283 @@
+uuid: d0347b96-ad05-4ceb-a9c6-331d9bc39699
```
Noticing this file is new.

joewhitsitt replied:
due to entity print

pyrello committed:
These files need to be split because of the private_content module.

Commit:
Require drupal/core-composer-scaffold:^9.2, no change

pyrello committed:
Updating private settings per content type.

Commit:
Require drupal/core-dev ^9.2

- Removing behat/mink-browserkit-driver (v1.3.4)
- Upgrading drupal/core-dev (9.1.3 => 9.2.0-beta2)
- Locking friends-of-behat/mink-browserkit-driver (v1.5.0)

Commit:
Export default config after manual sql:sync, cim then updb

richardbporter commented on config/default/system.advisories.yml:
```diff
@@ -0,0 +1,2 @@
+interval_hours: 6
+enabled: true
```
We may want to turn this off. https://www.drupal.org/node/3206473

joewhitsitt replied:
they have really messed with our demos as of late. "don't look at that giant security warning, instead look at this new button you can press."

richardbporter replied:
We have a lot of people getting notified through other means. We don't need hundreds of sites making outbound requests on cron.

joewhitsitt committed:
update v2 page.default

joewhitsitt committed:
MOAR config using the recommended method

pyrello committed:
Re-export of entity_form_display with new yaml format. Updated view to only allow the last year of results.

pyrello committed:
Hide private status on this content type.

Commit:
Disable system advisories

pyrello committed:
Updated to actual patch file.

joewhitsitt committed:
Merge branch 'main' into drupal-nine-two

pyrello committed:
Merge branch 'main' into drupal-nine-two

richardbporter committed:
Merge branch 'main' into drupal-nine-two
